src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -42,13 +42,11 @@
     /** @var array */
     private $dbParameters;
 
-    public function __construct(array $parameters, array $state = [], ?Logger $logger = null)
+    public function __construct(array $parameters, array $state, Logger $logger)
```

Comment:
state je teda v configu taky vždycky, jo? Měl jsem pocit, že nějak když se to spouští přímo přes CLI, tak ho mít nemusí...

Reply:
Sam o sobe by se ten Extractor poustet nemel. Cele by se to melo poustet pres Application, kde se konretni extraktor vytvari pres factory. Muzes mi sem linknout, co to spousti pres CLI?

Reply:
state do `Application` je optional, ale tady je to `Extractor` a to je vzdycky pod `Application`